tests/services_test.go

```diff
@@ -145,7 +145,7 @@ func (s *ServicesTestSuite) TestExecuteServiceErrStatus0() {
 		ID: 0,
 		Cause: rpc.Error{
 			ErrorCode:   1000,
-			Description: "transaction receipt has status 0 which indicates a transaction execution failure with error error",
+			Description: "transaction receipt has status 0 which indicates a transaction execution failure with error 0x6572726F72", // "error"
```

Comment:
Is this what the errors will look like now? I fail to see how this is more useful.

Reply:
it's not for you. it's for the client. do you enjoy seeing encrypted error messages a la
```
transaction receipt has status 0 which indicates a transaction execution failure with error \ufffd\ufffd\u001eQU^\ufffd\ufffd\ufffd\ufffdA\ufffd P\ufffd@\ufffd0\ufffd\ufffd49\ufffd?\ufffd\ufffd\ufffd\ufffd\ufffd\\\ufffd-\u0014\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0003\ufffdre\ufffd\ufffd\ufffd*\u00123\u0008\ufffd\ufffd\ufffd\ufffdD\u0016\ufffdfԈ\t\ufffd\ufffd\ufffd\"\u0015Xl\ufffd\u0007\ufffd\ufffdC"
```

Reply:
The PR description says that "This PR retains the original hex-encoded error for processing by clients."

But further down the stack, the hex-encoded error is now a part of a human-readable error message (not a machine-oriented one), so it will be harder to decode, right? It feels like we should either decode here (as it was before this PR) or pass the hex onwards in a special field so that it can later be processed easily and reliably.